astropy/io/fits/connect.py

```diff
@@ -236,7 +236,7 @@ def write_table_fits(input, output, overwrite=False):
         Whether to overwrite any existing file without warning.
     """
 
-    table_hdu = table_to_hdu(input)
+    table_hdu = table_to_hdu(input, character_as_bytes=True)
```

Comment:
This is deliberate - there are no situations I can think of under which we should expose ``character_as_bytes`` as a keyword argument in ``write_table_fits`` (unlike for reading)